ethers-contract/src/call.rs

```diff
@@ -211,3 +218,18 @@ where
             .map_err(ContractError::MiddlewareError)
     }
 }
+
+/// [`ContractCall`] can be turned into [`Future`] automatically with `.await`.
+/// Defaults to calling [`ContractCall::call`].
+impl<M, D> IntoFuture for ContractCall<M, D>
+where
+    M: Middleware + 'static,
+    D: Detokenize + 'static,
+{
+    type Output = Result<D, ContractError<M>>;
+    type IntoFuture = Pin<Box<dyn Future<Output = Self::Output>>>;
+
+    fn into_future(self) -> Self::IntoFuture {
+        Box::pin(async move { self.call().await })
+    }
```

Comment:
I'm not fond of excessive allocations, but it helps to hide an opaque future type.

Reply:
GATs may help avoid this one in the future, but because this is a static lifetime future, it may be impossible to avoid